### ISSUE TITLE
Add internal slots to MLOperand, MLActivation and basic algorithms

### ISSUE THREAD

**Issue by zolkis** [webmachinelearning/webnn#336]
In the validation steps for various operations (for instance Batch Norm), the following dependencies/references/algorithms need to be defined (this entry will be kept updated):
- **Internal slots** for MLOperand: 
    - [[type]] as MLOperandType
    - [[dimensions]] as `sequence<unsigned long>`
    - [[name]] as string (only for input operands)
    - [[data]] as ArrayBufferView (tensor of given dimensions)
    - [[builder]] the associated MLGraphBuilder
- Algorithm (or internal slot) to get the **_rank_** of the operand (for checking range for axis)
- Algorithm to internally **create** an MLOperand given |builder| and |descriptor|, used by op algorithms.
- Algorithm to internally **copy** an MLOperand, used by op algorithms.
- Algorithm to **check dimensions** given |dimensions| and |type|.
- **Internal slots** for MLActivation:
    - [[name]] as string, e.g. "relu", "sigmoid", "softmax", etc.
    - ? [[input]] as MLOperand (needed?)
    - ? [[output]] as MLOperand (intermediate output?)
- Algorithm to **create** an MLActivation, used by the op algorithms.

Related to the clarification of terms requested in #335.

**Comment by anssiko** [comment]
@zolkis given PR https://github.com/webmachinelearning/webnn/pull/400#issuecomment-1602228662 was closed please provide a record here of commit(s) and/or PR(s) that address this issue. When done, close this issue.

My recommendation is to link each PR with an issue using the [Fix keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) either in the PR description of the commit message to enable the associated issue(s) to be automatically closed upon merge.

**Comment by zolkis** [comment]
The content of #400, and previously, #337, has been agreed with @huningxin to be split among the PRs that will have dependencies to this content: 
#347, #348, #398 (#366), #365
